From 82a12ff23267746f230d39469b2d649001dfc3a9 Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <daniel.dragomir@windriver.com>
Date: Wed, 29 Apr 2026 20:28:59 +0300
Subject: [PATCH] add -std=gnu11 on sub-builds for GCC 15

GCC 15 defaults to -std=gnu23 where 'false' and 'bool' are reserved
keywords, breaking sub-builds that reconstruct their own KBUILD_CFLAGS
independently of the main kernel build which has gnu11 standard set:

 - arch/x86/realmode/rm/ uses REALMODE_CFLAGS
 - drivers/firmware/efi/libstub/ reconstructs KBUILD_CFLAGS from scratch
 - arch/x86/boot/compressed/ reconstructs KBUILD_CFLAGS from scratch

Upstream-Status: Pending

Signed-off-by: Daniel Dragomir <daniel.dragomir@windriver.com>
---
 arch/x86/Makefile                     | 1 +
 arch/x86/boot/compressed/Makefile     | 1 +
 drivers/firmware/efi/libstub/Makefile | 1 +
 3 files changed, 3 insertions(+)

diff --git a/arch/x86/Makefile b/arch/x86/Makefile
index 3419ffa2a350..2e01d61ad3b2 100644
--- a/arch/x86/Makefile
+++ b/arch/x86/Makefile
@@ -51,6 +51,7 @@ REALMODE_CFLAGS	:= -m16 -g -Os -DDISABLE_BRANCH_PROFILING -D__DISABLE_EXPORTS \
 REALMODE_CFLAGS += -ffreestanding
 REALMODE_CFLAGS += -fno-stack-protector
 REALMODE_CFLAGS += -Wno-address-of-packed-member
+REALMODE_CFLAGS += -std=gnu11
 REALMODE_CFLAGS += $(cc_stack_align4)
 REALMODE_CFLAGS += $(CLANG_FLAGS)
 export REALMODE_CFLAGS
diff --git a/arch/x86/boot/compressed/Makefile b/arch/x86/boot/compressed/Makefile
index 897f56533e6c..9cad4cc684d8 100644
--- a/arch/x86/boot/compressed/Makefile
+++ b/arch/x86/boot/compressed/Makefile
@@ -52,6 +52,7 @@ KBUILD_CFLAGS += -D__DISABLE_EXPORTS
 # Disable relocation relaxation in case the link is not PIE.
 KBUILD_CFLAGS += $(call cc-option,-Wa$(comma)-mrelax-relocations=no)
 KBUILD_CFLAGS += -include $(srctree)/include/linux/hidden.h
+KBUILD_CFLAGS += -std=gnu11
 
 # sev.c indirectly inludes inat-table.h which is generated during
 # compilation and stored in $(objtree). Add the directory to the includes so
diff --git a/drivers/firmware/efi/libstub/Makefile b/drivers/firmware/efi/libstub/Makefile
index 748781c25787..340756adac10 100644
--- a/drivers/firmware/efi/libstub/Makefile
+++ b/drivers/firmware/efi/libstub/Makefile
@@ -36,6 +36,7 @@ KBUILD_CFLAGS			:= $(cflags-y) -Os -DDISABLE_BRANCH_PROFILING \
 				   -D__NO_FORTIFY \
 				   -ffreestanding \
 				   -fno-stack-protector \
+				   -std=gnu11 \
 				   $(call cc-option,-fno-addrsig) \
 				   -D__DISABLE_EXPORTS
 
-- 
2.39.5

